django/forms/formsets.py

```diff
@@ -127,11 +154,9 @@ def total_form_count(self):
     def initial_form_count(self):
         """Returns the number of forms that are required in this FormSet."""
         if self.is_bound:
-            return self.management_form.cleaned_data[INITIAL_FORM_COUNT]
-        else:
-            # Use the length of the initial data if it's there, 0 otherwise.
-            initial_forms = len(self.initial) if self.initial else 0
-        return initial_forms
+            return self._extract_initial_form_count()
+        # Use the length of the initial data if it's there, 0 otherwise.
+        return len(self.initial) if self.initial else 0
```

Comment:
This used to be in an `else` clause, now it is not. Meaning that calling `self._extract_initial_form_count()` was pointless, because it will be overridden anyway.